arangod/Cluster/ServerState.cpp

```diff
@@ -319,14 +319,26 @@ ServerState::Mode ServerState::setServerMode(ServerState::Mode value) {
 }
 
 static std::atomic<bool> _serverstate_readonly(false);
+static std::atomic<bool> _license_readonly(false);
 
 bool ServerState::readOnly() {
-  return _serverstate_readonly.load(std::memory_order_acquire);
+  return _serverstate_readonly.load(std::memory_order_acquire) ||
+    _license_readonly.load(std::memory_order_acquire);
 }
 
 /// @brief set server read-only
-bool ServerState::setReadOnly(bool ro) {
-  return _serverstate_readonly.exchange(ro, std::memory_order_release);
+bool ServerState::setReadOnly(ReadOnlyMode ro) {
+  if(ro == API_FALSE) {
+    _serverstate_readonly.exchange(false, std::memory_order_release);
+  } else if (ro == API_TRUE) {
+    _serverstate_readonly.exchange(true, std::memory_order_release);
+  } else if (ro == LICENSE_FALSE) {
+    _license_readonly.exchange(false, std::memory_order_release);
+  } else if (ro == LICENSE_TRUE) {
+    _license_readonly.exchange(true, std::memory_order_release);
+  }
+  return _serverstate_readonly.load(std::memory_order_acquire) ||
+    _license_readonly.load(std::memory_order_acquire);
```

Comment:
This is a behavior change: the previous implementation returned the _previous_ value of the read-only flag, the new implementation returns the new value.
I think it makes sense to restore the old behavior so everything remains fully downwards-compatible.